### ISSUE TITLE
Failure during docker/spark build

### ISSUE THREAD

**Issue by tannousmarc** [Quetzal-RDF/quetzal#9]
Trying to setup the project on OSX, in a Ubuntu VM as per the getting started guide. Some assumptions I made:

- When you mention adding quetzalRDF as the shared folder, I added this whole repo.
- Building the docker image from /media/sf_quetzal/com.ibm.research.quetzal.core/docker/spark by running ``sudo docker build --no-cache --rm -t "ibmresearch/quetzal_spark" .``

During this command, one of the intermediate steps fails so the whole thing looks like it's failing, getting back: 

``The command '/bin/sh -c curl -O http://ftp.gnu.org/gnu/coreutils/coreutils-8.24.tar.xz     && tar -xJf coreutils-8.24.tar.xz 	&& export FORCE_UNSAFE_CONFIGURE=1 	&& cd ./coreutils-8.24 	&& ./configure 	&& make 	&& make install && cd .. 	&& curl -O http://ftp.gnu.org/gnu/gawk/gawk-4.1.3.tar.xz     && tar -xJf gawk-4.1.3.tar.xz 	&& cd ./gawk-4.1.3 	&& ./configure 	&& yum -y install wget 	&& wget --no-cookies --no-check-certificate --header "Cookie: oraclelicense=accept-securebackup-cookie" "http://download.oracle.com/otn-pub/java/jdk/$JAVA_VERSION-$BUILD_VERSION/jdk-$JAVA_VERSION-linux-x64.rpm" -O /tmp/jdk-8-linux-x64.rpm 	&& yum -y install /tmp/jdk-8-linux-x64.rpm 	&& alternatives --install /usr/bin/java jar /usr/java/latest/bin/java 200000 	&& alternatives --install /usr/bin/javaws javaws /usr/java/latest/bin/javaws 200000 	&& alternatives --install /usr/bin/javac javac /usr/java/latest/bin/javac 200000 	&& export JAVA_HOME=/usr/java/latest 	&& make 	&& make install 	&& export PATH=/usr/local/bin:$PATH && cd .. 	&& curl -O http://mirror.sdunix.com/apache/maven/maven-3/3.3.3/binaries/apache-maven-3.3.3-bin.tar.gz 	&& tar -xzf apache-maven-3.3.3-bin.tar.gz 	&& yum -y install git		&& mkdir /data' returned a non-zero code: 8
``

Looking back at what happened, it seems like fetching jdk fails, with the following log:

``--2019-03-06 11:51:40--  http://download.oracle.com/otn-pub/java/jdk/8u31-b13/jdk-8u31-linux-x64.rpm?AuthParam=1551891220_93972698f20d7de1573ddea29a4c6071
Connecting to download.oracle.com|92.122.164.143|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://download.oracle.com/otn-pub/java/jdk/8u31-b13/jdk-8u31-linux-x64.rpm?AuthParam=1551891220_93972698f20d7de1573ddea29a4c6071 [following]
--2019-03-06 11:51:40--  https://download.oracle.com/otn-pub/java/jdk/8u31-b13/jdk-8u31-linux-x64.rpm?AuthParam=1551891220_93972698f20d7de1573ddea29a4c6071
Connecting to download.oracle.com|92.122.164.143|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-03-06 11:51:43 ERROR 404: Not Found.
``

Is there any way to get this working?

**Comment by ksrinivs64** [comment]
Can you try:
ENV JAVA_VERSION 8u201
ENV BUILD_VERSION b09

The docker image is pretty old and when I try it locally I am getting errors well before the jdk install.  If your system gets further, this may be one way to bypass the problem.

**Comment by tannousmarc** [comment]
Modified the Dockerfile to contain your ENV variables, but building still failed with the same error.

I navigated to the oracle webpage, got the URL of 8u201 and hardcoded it into the Dockerfile, which appears to work.

Then, the build failed when downloading maven (the mirror in the Dockerfile returns a 404). I hardcoded another mirror to the same version in the file.

At this point, building again told me ``/var/cache/yum/x86_64/6/base/packages`` ran out of space. In fact, the VM had ran out of space (initialised it with 10GB, as per the instructions). Resized it to 20GB.

Ran the command again, and it failed during some intermediary step:

``make  install-data-hook
make[4]: Entering directory `/gawk-4.1.3/extension'
for i in filefuncs.la fnmatch.la fork.la inplace.la ordchr.la readdir.la readfile.la revoutput.la revtwoway.la rwarray.la testext.la time.la ; do \
		rm -f /usr/local/lib/gawk/$i ; \
	done
make[4]: Leaving directory `/gawk-4.1.3/extension'
make[3]: Leaving directory `/gawk-4.1.3/extension'
make[2]: Leaving directory `/gawk-4.1.3/extension'
make[1]: Leaving directory `/gawk-4.1.3/extension'
Making install in test
make[1]: Entering directory `/gawk-4.1.3/test'
make[2]: Entering directory `/gawk-4.1.3/test'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/gawk-4.1.3/test'
make[1]: Leaving directory `/gawk-4.1.3/test'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (35) SSL connect error
``

At this point I've put in some time into trying to get this to work, and it doesn't seem likely. Is there any other way I could get this project to work on OSX? Or even Windows? It would come in very handy with my dissertation.
